core-db/src/main/java/io/novafoundation/nova/core_db/migrations/52_53_AddConnectionStateToChains.kt

```diff
@@ -0,0 +1,25 @@
+package io.novafoundation.nova.core_db.migrations
+
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
+import io.novafoundation.nova.core_db.model.chain.ChainLocal
+
+val AddConnectionStateToChains_53_54 = object : Migration(53, 54) {
+    override fun migrate(database: SupportSQLiteDatabase) {
+        val defaultConnectionState = ChainLocal.Default.CONNECTION_STATE_DEFAULT
+
+        database.execSQL("ALTER TABLE chains ADD COLUMN connectionState TEXT NOT NULL DEFAULT '$defaultConnectionState'")
+
+        // Enable full for chains that have some balance synced
+        database.execSQL(
+            """
+            UPDATE chains SET connectionState = 'FULL_SYNC'
```

Comment:
From this logic it mean that noRuntime (evm) netowrks might work as light_sync or full_sync. Probably it is better to manually set light_sync or full_sync for them.